FE-Atomic Button 요소들 개발

do02reen24 opened the pull request:
### 📕 Issue Number

linked issue #1
<br/>

### 📗 작업 내역

> 구현 내용 및 작업 했던 내역

-  atoms/button 개발
    - SquircleShortButton
    - RoundShortButton
    - RoundButton
    - RoundLongButton
    - TextButton

<br/>

### 📘 작업 유형

- 신규 기능 추가

<br/>

### 📝 PR 특이 사항

> PR을 볼 때 주의깊게 봐야하거나 말하고 싶은 점

- storybook에서 좌우 너비를 바꾸면서 버튼의 크기를 변경하면서 확인 부탁드립니다.
   - 일부 버튼의 경우에는 **%** 를 적용하였습니다.
   - 버튼의 크기나 여백이 적당한지 봐주세요.
- 각 컴포넌트에서 현재 전달되고 있는 props가 과하거나 부족하지 않은지 확인 부탁드립니다.
- 아직 초반이라 코딩스타일이 다른 부분이 있을 것 같습니다. 개선할 부분이 있다면 의견 남겨주세요! 반영하도록 하겠습니다 😊
- favicon을 꿀벌🐝모양으로 변경해보았습니다. 어떤지 피드백 부탁드립니다 ㅎㅎ

<br/><br/>

do02reen24 committed:
feat : favicon 변경
- favicon을 꿀벌 모양으로 변경함

do02reen24 committed:
feat : RoundShortButton atom 개발
- 기획에 맞춰 defaultProps를 설정해줌
- min, max width를 px로 걸어주고 그 외는 rem으로 설정함

do02reen24 committed:
feat : RoundButton, RoundLongButton 개발
- RoundShortButton과 유사하게 그 외 버튼을 개발함
- Props를 통해 넘겨받는 요소들은 하단쪽으로 모아서 선언해줌

do02reen24 committed:
feat : SquircleShortButton, TextButton 개발
- SquircleShortButton, TextButton 을 개발함
- SquircleShortButton에 사용할 lightGray를 color에 추가함

Zigje9 commented on client/src/components/atoms/button/RoundButton/RoundButton.tsx:
```diff
@@ -0,0 +1,39 @@
+  onClick: undefined,
+};
+
+const Button = styled.button<Props>`
+  width: 45%;
+  font-size: 1rem;
+  border: 0px;
+  border-radius: 25px;
+  cursor: pointer;
+  padding: 0.25rem 0;
+  background-color: ${(props) => props.backColor};
+  color: ${(props) => props.color};
+`;
+
```
완벽한 css 인거 같습니다. rem 은 최상위를 따라가고 em 은 상위를 따라간다는데. em 을 선택하는것이 좀 더 안전하지 않을까요?

do02reen24 replied:
페이지 전체에서 버튼이 일관성을 갖게하기 위해서 rem을 사용했습니다. em으로 바꾸고 일관성을 가져야하는 경우 페이지 상위 컴포넌트마다 동일한 값을 지정해주는 것도 괜찮을 것 같네요.

Zigje9 commented on client/src/components/atoms/button/RoundButton/RoundButton.tsx:
```diff
@@ -0,0 +1,39 @@
+import myColor from '@theme/color';
+
+interface Props {
+  backColor?: string;
+  color?: string;
```
background-color 나 color 의 props 를 넘겨주지 않는 상황이 있을까요? 갑자기 궁금해졌습니다!..

do02reen24 replied:
글씨 (배경이 투명한) 버튼 같은 경우에는 background-color가 interface에 선언되어 있지 않습니다 ㅎㅎ

Zigje9 reviewed (approved):
디자인과 css 감각이 너무 뛰어나신거 같네요~ 항상 배워갑니다. 고생하셨습니다.

leejiwoo2002 commented on client/src/components/atoms/button/RoundButton/RoundButton.tsx:
```diff
@@ -0,0 +1,39 @@
+};
+
+const Button = styled.button<Props>`
+  width: 45%;
```
어제 의논한 대로, 개발하면서 %, px 단위는 일단 만들어 보고 나중에 같이 조정해보시죠!

do02reen24 replied:
화면에 어떻게 들어갈지 감이 안잡히네요🤔 분자와 유기체 단위를 만들면서 지속적으로 개선시켜 나가는 것도 좋을 것 같습니다

do02reen24 committed:
refactor :  PR 리뷰 반영
- 스토리북 componenet [ ] 형태로 변경
- 스토리북 componenet return 타입 지정
- backColor 이름 backgroundColor로 변경(통일화)

do02reen24 committed:
Merge branch 'develop' into FE-feat-atoms-doyeon

JunYoung7 commented on client/src/components/atoms/button/RoundButton/RoundButton.tsx:
```diff
@@ -0,0 +1,39 @@
+interface Props {
+  backgroundColor?: string;
+  color?: string;
+  children: React.ReactChild;
```
ReactChild가 혹시 무슨 의미를 가지고 있고 어떤 역할을 수행할까요??

do02reen24 replied:
interface에 타입을 선언해줘야하는데 children에는 ReactChild라는 타입을 선언해준 것이고, children은 컴포넌트 안에 들어갈 요소들을 전달해주기 위해 사용합니다.

JunYoung7 reviewed (commented):
고생하셨습니다 도연님!
